### ISSUE TITLE
Problems in Unseen Splits

### ISSUE THREAD

**Issue by KennithLi** [TheShadow29/zsgnet-pytorch#10]
Thank you for your great work. Recently, I am interested in the zero-shot setting for referring expression comprehension task and have read your paper. I meet some problems when I reproduced the seen/unseen splits for Flickr30k Entities and Visual Genome datasets. Can you share the process you made the seen/unseen splits and the related codes? I want to know the exact phrases or objects you did the splits. Many thanks and looking forward to hearing from you.

**Comment by TheShadow29** [comment]
@KennithLi Thank you for your interest in the work. 

Could you briefly describe the problem you face in reproducing the results? 

For flickr30k, case 0 (as described in paper) is already provided. 

For case1, we just remove instances of "other" for seen, and use only those instances in the unseen. We then remove a portion of the images from seen which also contain unseen objects (to avoid any spill from train to test).

For visual genome, we extract out the nouns (directly available from VG iirc), and then apply Glove embedding, followed by K-means cluster with k=20. For each cluster, we order by frequency (descending order) and split into two halves, first being seen and the second being unseen.

For case2 test set, we ensure there is no seen classes belonging to the same cluster of unseen classes.
For case test set, we ensure there is at least one seen class belonging to the same cluster of unseen class.
